MessageQueue/ServerStatisticsReceiver.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Text;
+using System.Text.Json;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using ServerStatistics.Extensions;
+using ServerStatistics.Models;
+using ServerStatistics.Services;
+using SignalREndpoint;
+
+namespace MessageQueue
+{
+    public class ServerStatisticsReceiver : RabbitMqConnector, IMessageReceiver
+    {
+        private string _queueName;
+        private IServerStatisticsService _statisticsService;
+        private ServerStatisticsDTO _serverStatistics;
+        private IAlertSender _alertSender;
+        private bool _hasPreviousStatistics = false;
+        private ServerStatisticsDTO _previousStatistics;
+
+        public ServerStatisticsReceiver(
+            string connectionName,
+            string exchangeName,
+            string queueName,
+            string routingKey,
+            IServerStatisticsService statisticsService,
+            IAlertSender alertSender) : base(connectionName)
+        {
+            _channel.ExchangeDeclare(exchangeName, ExchangeType.Topic);
```

Comment:
seems like these three lines here in the sender class are common and can be just moved to `RabbitMqConnector` constructor.